code/game/objects/items/weapons/extinguisher.dm

```diff
@@ -46,6 +46,17 @@
 	user << "The safety is [safety ? "on" : "off"]."
 	return
 
+/obj/item/weapon/extinguisher/attack(mob/M, mob/user)
+	if(user.a_intent == "help")
+		// If we're in help intent, don't bash anyone with the
+		// extinguisher
+		user.visible_message("[user] targets [M] with \the [src]", "<span class='info'>You target [M] with \the [src].</span>")
+		// FIXME the fire extinguisher isn't triggering, don't know

```

Comment:
call afterattack because your returning before calling the parent.
afterattack(m, user) should do it